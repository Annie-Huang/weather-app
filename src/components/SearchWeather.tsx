import React, {ChangeEvent, useState} from 'react';
import './SearchWeather.css';

const SearchWeather = ({city, setCity}) => {
  const [searchCity, setSearchCity] = useState(city);

  const handleSubmitData = () => {
    console.log('annieannie, searchCity=', searchCity);
    setCity(searchCity);
  }

  return (
    <form className='my-3'>
      <label htmlFor="city" className="form-label">Search Weather for your city</label>
      <input type="text" className="form-control" id="city" name="city" value={searchCity} onChange={(e) => setSearchCity(e.target.value)}/>
      city: {city}
      <br />

      Search city: {searchCity}
      <br />

      {/*<div className="form-check mt-2">*/}
      {/*  <input className="form-check-input" type="radio" name="weatherUnit" id="Celsius" checked/>*/}
      {/*    <label className="form-check-label" htmlFor="Celsius">*/}
      {/*      Celsius*/}
      {/*    </label>*/}
      {/*</div>*/}
      {/*<div className="form-check mb-2">*/}
      {/*  <input className="form-check-input" type="radio" name="weatherUnit" id="Fahrenheit" />*/}
      {/*    <label className="form-check-label" htmlFor="Fahrenheit">*/}
      {/*      Fahrenheit*/}
      {/*    </label>*/}
      {/*</div>*/}

      {/*<button type="submit" className="btn btn-primary" onClick={handleSubmitData}>Search</button>*/}
      {/*<button className="btn btn-primary" onClick={handleSubmitData}>Search</button>*/}
      <button className="btn btn-primary" onClick={() => setCity(searchCity)}>Search</button>
    </form>
  );
};

export default SearchWeather;

/*
<input class="form-control form-control-lg" type="text" placeholder=".form-control-lg" aria-label=".form-control-lg example">
<input class="form-control" type="text" placeholder="Default input" aria-label="deafult input example">
<input class="form-control form-control-sm" type="text" placeholder=".form-control-sm" aria-label=".form-control-sm example">
 */

