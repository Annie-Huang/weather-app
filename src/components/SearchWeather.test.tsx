import React from 'react';
import { mount, shallow } from 'enzyme';
import SearchWeather, { ISearchWeatherProp } from './SearchWeather';

test('SearchWeather - snapshot', () => {
  const props: ISearchWeatherProp = {
    city: 'cityStr',
    setCity: jest.fn(),
    units: 'unitsStr',
    setUnits: jest.fn(),
  };

  const wrapper = shallow(<SearchWeather {...props} />);
  expect(wrapper).toMatchSnapshot();
});

test('SearchWeather - change radio button will make setUnits call', () => {
  const props: ISearchWeatherProp = {
    city: 'cityStr',
    setCity: jest.fn(),
    units: 'unitsStr',
    setUnits: jest.fn(),
  };

  const wrapper = shallow(<SearchWeather {...props} />);
  wrapper
    .find('input[type="radio"]')
    .at(1)
    .simulate('change', { target: { checked: true } });
  expect(props.setUnits).toHaveBeenCalledWith('I');
  wrapper
    .find('input[type="radio"]')
    .at(0)
    .simulate('change', { target: { checked: true } });
  expect(props.setUnits).toHaveBeenCalledWith('M');
});

test('SearchWeather - change input and click search button will make setCity call', () => {
  const props: ISearchWeatherProp = {
    city: 'cityStr',
    setCity: jest.fn(),
    units: 'unitsStr',
    setUnits: jest.fn(),
  };

  const cityToSearch = 'mitcham';

  // Have to use full mount otherwise it will throw:
  // ShallowWrapper::instance() can only be called on the root exception
  const wrapper = mount(<SearchWeather {...props} />);

  // wrapper.find('input[type="text"]').instance<HTMLInputElement>().value = cityToSearch;   // Very annoying this short form doesn't work
  wrapper.find('input[type="text"]').simulate('change', { target: { value: cityToSearch } });
  wrapper.find('button').simulate('click');
  expect(props.setCity).toHaveBeenCalledWith(cityToSearch);
});
