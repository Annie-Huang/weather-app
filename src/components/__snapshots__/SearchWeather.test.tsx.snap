// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchWeather - snapshot 1`] = `
<form
  className="my-3 d-flex flex-column justify-content-center"
>
  <label
    className="form-label"
    htmlFor="city"
  >
    Search Weather for your city
  </label>
  <input
    aria-label="search city"
    className="form-control mb-2"
    id="city"
    name="city"
    onChange={[Function]}
    type="text"
    value="cityStr"
  />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Search
  </button>
  <div
    className="d-flex justify-content-center"
  >
    <div
      className="form-check mt-2 mr-4"
      key="0"
    >
      <input
        checked={false}
        className="form-check-input"
        id="Celsius"
        name="weather-unit"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="form-check-label"
        htmlFor="Celsius"
      >
        Celsius
      </label>
    </div>
    <div
      className="form-check mt-2 mr-4"
      key="1"
    >
      <input
        checked={false}
        className="form-check-input"
        id="Fahrenheit"
        name="weather-unit"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="form-check-label"
        htmlFor="Fahrenheit"
      >
        Fahrenheit
      </label>
    </div>
  </div>
</form>
`;
